app/data/smiles.py

```diff
@@ -40,22 +41,34 @@ def construct_smiles(csv):
   return smiles, include_mpo
 
 def filter_smiles(good_smiles, smiles):
-  return {smi: data for (smi, data) in smiles.items() if smi in good_smiles}
+  return {smi: data for (smi, data) in smiles.items() if data['smart'] in convert_array_of_smarts_to_smiles(good_smiles)}
 
 def convert(mol_smile):
```

Comment:
nit: can we rename this function since there are now a lot of convert functions?